src/types.ts

```diff
@@ -1,34 +1,21 @@
 import * as _ from 'lodash';
 import * as Web3 from 'web3';
 
-// Utility function to create a K:V from a list of strings
-// Adapted from: https://basarat.gitbooks.io/typescript/content/docs/types/literal-types.html
-export interface StringEnum {
-    [key: string]: string;
-}
-function strEnum(values: string[]): StringEnum {
-    return _.reduce(values, (result, key) => {
-        result[key] = key;
-        return result;
-    }, Object.create(null));
-}
-
-export const ZeroExError = strEnum([
-    'CONTRACT_DOES_NOT_EXIST',
-    'EXCHANGE_CONTRACT_DOES_NOT_EXIST',
-    'UNHANDLED_ERROR',
-    'USER_HAS_NO_ASSOCIATED_ADDRESSES',
-    'INVALID_SIGNATURE',
-    'CONTRACT_NOT_DEPLOYED_ON_NETWORK',
-    'ZRX_NOT_IN_TOKEN_REGISTRY',
-    'INSUFFICIENT_ALLOWANCE_FOR_TRANSFER',
-    'INSUFFICIENT_BALANCE_FOR_TRANSFER',
-    'INSUFFICIENT_ETH_BALANCE_FOR_DEPOSIT',
-    'INSUFFICIENT_WETH_BALANCE_FOR_WITHDRAWAL',
-    'INVALID_JUMP',
-    'OUT_OF_GAS',
-]);
-export type ZeroExError = keyof typeof ZeroExError;
+export enum ZeroExError {
+    CONTRACT_DOES_NOT_EXIST = 'CONTRACT_DOES_NOT_EXIST',
+    EXCHANGE_CONTRACT_DOES_NOT_EXIST = 'EXCHANGE_CONTRACT_DOES_NOT_EXIST',
+    UNHANDLED_ERROR = 'UNHANDLED_ERROR',
+    USER_HAS_NO_ASSOCIATED_ADDRESSES = 'USER_HAS_NO_ASSOCIATED_ADDRESSES',
+    INVALID_SIGNATURE = 'INVALID_SIGNATURE',
+    CONTRACT_NOT_DEPLOYED_ON_NETWORK = 'CONTRACT_NOT_DEPLOYED_ON_NETWORK',
+    ZRX_NOT_IN_TOKEN_REGISTRY = 'ZRX_NOT_IN_TOKEN_REGISTRY',
+    INSUFFICIENT_ALLOWANCE_FOR_TRANSFER = 'INSUFFICIENT_ALLOWANCE_FOR_TRANSFER',
+    INSUFFICIENT_BALANCE_FOR_TRANSFER = 'INSUFFICIENT_BALANCE_FOR_TRANSFER',
+    INSUFFICIENT_ETH_BALANCE_FOR_DEPOSIT = 'INSUFFICIENT_ETH_BALANCE_FOR_DEPOSIT',
+    INSUFFICIENT_WETH_BALANCE_FOR_WITHDRAWAL = 'INSUFFICIENT_WETH_BALANCE_FOR_WITHDRAWAL',
+    INVALID_JUMP = 'INVALID_JUMP',
+    OUT_OF_GAS = 'OUT_OF_GAS',
+}
```

Comment:
Let's follow the suggested convention from the Typescript release notes and use capitalized, camel-case names for the string enum entries, and have the underlying values still be cap, snake-case.